commit 09e8d0275efea92f2f4c5d5ed657b3d45c62bde7 (refs/changes/84/1184/1)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2016-12-27T08:34:49-08:00 (2 years, 9 months ago)
    
    joyent/node-sdc-clients#10 Want support for sync parameter for vmapi.getVM

diff --git a/lib/vmapi.js b/lib/vmapi.js
index 8447397..02792ea 100644
--- a/lib/vmapi.js
+++ b/lib/vmapi.js
@@ -336,6 +336,8 @@ VMAPI.prototype.getVm = function (params, options, callback) {
         query.owner_uuid = params.owner_uuid;
     if (params.fields)
         query.fields = params.fields;
+    if (params.sync)
+        query.sync = 'true';
 
     var opts = { path: format('/vms/%s', params.uuid), query: query };
     if (options) {
diff --git a/test/vmapi.test.js b/test/vmapi.test.js
index 4006b19..1449d7e 100644
--- a/test/vmapi.test.js
+++ b/test/vmapi.test.js
@@ -372,6 +372,18 @@ test('vmapi', function (tt) {
         });
     });
 
+    tt.test(' sync vm', function (t) {
+        var SYNC_QUERY = {
+            uuid: ZONE,
+            sync: 'true'
+        };
+        vmapi.getVm(SYNC_QUERY, function (err, vm) {
+            t.ifError(err);
+            t.ok(vm);
+            t.end();
+        });
+    });
+
 
     tt.test(' find headnode', function (t) {
         cnapi.listServers(function (err, servers) {
@@ -1181,4 +1193,4 @@ test('vmapi', function (tt) {
         napi.close();
         t.end();
     });
-});
\ No newline at end of file
+});
